third_party/xla/xla/stream_executor/rocm/rocm_blas.cc

```diff
@@ -918,8 +918,8 @@ absl::StatusOr<AllocateStridedResult<T>> AllocateStridedBuffer(
   res.device_mem = DeviceMemory<MAPPED_T>(batch_matrix_bytes);
   res.reallocated = true;
   if (copy_data) {
-    return ReorganizeMemory(stream, &res.device_mem, raw_ptrs, batch_count,
-                            batch_stride, true);
+    TF_RETURN_IF_ERROR(ReorganizeMemory(stream, &res.device_mem, raw_ptrs,
+                       batch_count, batch_stride, true));
```

Comment:
Thanks for fixing this! could you upstream this return checking as well, please?

Reply:
Yes, I will do that.

Reply:
Yes, good catch! It seems that this bug occurred after my rocm blas refactoring. I do not have any good explanation why it's worked previously except that probably AllocateStridedBuffer always returned earlier [here](https://github.com/ROCmSoftwarePlatform/xla/blob/bd0f5a009be858826ba0a2390df40aeb840a3f14/xla/stream_executor/rocm/rocm_blas.cc#L978). Hence ReorganizeMemory has never been called..

Reply:
Also, if this `ReorganizeMemory()` needs to be checked, this part maybe also need to add as well I think? https://github.com/openxla/xla/blob/main/xla/stream_executor/rocm/rocm_blas.cc#L954-L957

Reply:
That, second, call to ReorganizeMemory is inside ROCMBlas::DoBlasGemmBatchedInternal and it returns absl::Status (not absl::StatusOr), thus I believe it is not needed.

Reply:
this patch is upstream https://github.com/openxla/xla/pull/10204